docs/concepts/performance.md

```diff
@@ -0,0 +1,182 @@
+| Inbound bandwidth  | 60 KBps | 60 KBps  | 60 KBps   | 60 KBps  | 60 KBps   | 60 KBps    | 60 KBps    |
+| Outbound messages per second | 3,000 | 6,000 | 15,000 | 30,000 | 60,000 | 150,000 | 300,000 |
+| Outbound bandwidth | 6 MBps   | 12 MBps  | 30 MBps  | 60 MBps  | 120 MBps  | 300 MBps   | 600 MBps   |
+
+##### Small group
+
+The routing cost is significant for sending message to many small groups. Currently, the Azure Web PubSub Service implementation hits the routing cost limit at Unit 50. Adding more CPU and memory doesn't help, so Unit100 can't improve further by design. If you need more inbound bandwidth, contact customer support.
+
+|   Send to small group     | Unit1 | Unit2 | Unit5  | Unit10 | Unit20  | Unit50 | Unit100 |
+|---------------------------|-------|-------|--------|--------|--------|--------|---------|
+| Connections               | 1,000 | 2,000 | 5,000  | 10,000 | 20,000 | 50,000 | 100,000 |
+| Group member count        | 10    | 10    | 10     | 10     | 10     | 10     | 10 |
+| Group count               | 100   | 200   | 500    | 1,000  | 2,000  | 5,000  | 10,000 |
+| Inbound messages per second  | 400 | 800  | 2,000 | 4,000 | 8,000 | 15,000 | 15,000 |
+| Inbound bandwidth  | 800 KBps | 1.6 MBps | 4 MBps    | 8 MBps    | 16 MBps   | 30 MBps  | 30 MBps   |
+| Outbound messages per second | 4,000 | 8,000 | 20,000 | 40,000 | 80,000 | 150,000 | 150,000 |
+| Outbound bandwidth | 8 MBps   | 16 MBps | 40 MBps   | 80 MBps    | 160 MBps  | 300 MBps | 300 MBps |
+
+### Triggering Cloud Event 
+Service delivers client events to the upstream webhook using the [CloudEvents HTTP protocol](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md).
+
+<img src="../images/upstream.png" alt="upstream" style="zoom:60%;" />
+
+For every event, it formulates an HTTP POST request to the registered upstream and expects an HTTP response. 
+
+> [!NOTE]
+> Web Pub Sub also supports HTTP 2.0 for upstream events delivering. The below result is tested using HTTP 1.1 . If your app server supports HTTP 2.0, the performance will be better. 
+
+#### Echo
+
+In this case, the app server writes back the original message back in the http response. The behavior of **echo** determines that the maximum inbound bandwidth is equal to the maximum outbound bandwidth. For details, see the following table.
+
+|       Echo                        | Unit1 | Unit2 | Unit5 | Unit10 | Unit20 | Unit50 | Unit100 |
+|-----------------------------------|-------|-------|-------|--------|--------|--------|---------|
+| Connections                       | 1,000 | 2,000 | 5,000 | 10,000 | 20,000 | 50,000 | 100,000 |
+| Inbound/outbound messages per second | 500 | 1,000 | 2,500 | 5,000 | 10,000 | 25,000 | 50,000 |
+| Inbound/outbound bandwidth | 1 MBps  | 2 MBps  | 5 MBps | 10 MBps  | 20 MBps  | 50 MBps | 100 MBps  |
+
+
+
+### Rest API
+
+Sending high-density messages through the REST API is not as efficient as using WebSocket. 
```

Comment:
We don't have a `using WebSocket` scenario, so maybe instead of the comparison, explain a little bit when to use REST API and simply show the performance test result below.

Reply:
Is there a doc or swagger introducing our rest API?

Reply:
https://docs.microsoft.com/en-us/rest/api/webpubsub/